src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -811,6 +811,46 @@ void NonClientIslandWindow::OnApplicationThemeChanged(const ElementTheme& reques
     _theme = requestedTheme;
 }
 
+// Method Description:
+// - Enable or disable borderless mode. When entering borderless mode, we'll
+//   need to manually hide the entire titlebar.
+// - See also IslandWindow::_SetIsBorderless, which does similar, but different work.
+// Arguments:
+// - borderlessEnabled: If true, we're entering borderless mode. If false, we're leaving.
+// Return Value:
+// - <none>
+void NonClientIslandWindow::_SetIsBorderless(const bool borderlessEnabled)
+{
+    _borderless = borderlessEnabled;

```

Comment:
Should we be testing on the way in that the state is actually changing before dispatching the window messages? Is there a way to set borderless = true on top of an already borderless window and dispatch a bunch of window messages that will make things jump around but end up back where they were in a weird way?